backends/sqs/backend.go

```diff
@@ -411,11 +415,18 @@ func (s *Backend) PublishMessages(ctx context.Context, in *rpc.PublishMessagesRe
 				StringValue: &pointerVal,
 			}
 		}
-		entries = append(entries, &sqs.SendMessageBatchRequestEntry{
+		entry := &sqs.SendMessageBatchRequestEntry{
 			Id:                &strIdx,
 			MessageAttributes: attrs,
 			MessageBody:       &message.Data,
-		})
+		}
+		if in.QueueId.Type == rpc.QueueId_Fifo {
+			dedupId := creaeteMessageDedupId()
+			entry.MessageDeduplicationId = &dedupId
```

Comment:
I just read some aws doc on this, it looks like the dedupId is need in some scenarios but not all. Do we know that our customer's use case require that? (not saying we shouldn't use the dedupId, just want to understand). 
Also, if we do need a dedupId, why inventing one instead of using a uuid?